src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionWorker.cs

```diff
@@ -101,5 +103,19 @@ await using (_processingCts.Token.Register(batch.SetCancel))
                 throw;
             }
         }
+
+        protected override (bool ShouldTryToReconnect, ServerNode NodeRedirectTo) HandleSubscriptionClosedException(SubscriptionClosedException sce)
+        {
+            if (sce.ClosedDueNoDocsLeft)
+            {
+                ClosedDueNoDocsLeft = true;
+                return (false, null);
+            }
+
+            if (sce.CanReconnect)
+                return (true, null);
+
+            return (false, null);
```

Comment:
`return base.HandleSubscriptionClosedException(sce)`

Reply:
cannot do since we don't pass the token anymore